runtime/autoload/health/provider.vim

```diff
@@ -91,7 +91,7 @@ function! s:download(url) abort
   if executable('curl')
     let rv = s:system(['curl', '-sL', a:url], '', 1, 1)
     return s:shell_error ? 'curl error with '.a:url.': '.s:shell_error : rv
-  elseif executable('python')
+  elseif executable(a:python)
```

Comment:
Actually, I reverted this change. `curl` is the main expectation, `python` is a fallback. This is only a download utility, not related to general python env sniffing.